confmap/provider.go

```diff
@@ -116,6 +122,32 @@ func WithRetrievedClose(closeFunc CloseFunc) RetrievedOption {
 	}
 }
 
+func withStringRepresentation(stringRepresentation string) RetrievedOption {
+	return func(settings *retrievedSettings) {
+		settings.stringRepresentation = stringRepresentation
+		settings.isSetString = true
+	}
+}
+
+// NewRetrievedFromYAML returns a new Retrieved instance that contains the deserialized data from the yaml bytes.
+// * yamlBytes the yaml bytes that will be deserialized.
+// * opts specifies options associated with this Retrieved value, such as CloseFunc.
+func NewRetrievedFromYAML(yamlBytes []byte, opts ...RetrievedOption) (*Retrieved, error) {
+	var rawConf any
+	if err := yaml.Unmarshal(yamlBytes, &rawConf); err != nil {
+		return nil, err
+	}
+
+	switch v := rawConf.(type) {
+	case string:
+		opts = append(opts, withStringRepresentation(v))
+	case int, int32, int64, float32, float64, bool:
+		opts = append(opts, withStringRepresentation(string(yamlBytes)))
```

Comment:
Are there test cases we can add for this?

Reply:
The end to end tests with `TargetFieldInlineString` test this. What kind of tests would you like to see?

Reply:
Ah, sorry, I think I got this wrong. You want to test this with something that does not have one of those types, right?

Reply:
If the e2e test covers this I'm good.